src/test/resources/com/puppycrawl/tools/checkstyle/checks/sizes/parameternumber/InputParameterNumberIgnoreAnnotatedByFullyQualifiedClassName.java

```diff
@@ -0,0 +1,62 @@
+/*
+ParameterNumber
+max = 2
+ignoreAnnotatedBy = java.lang.Deprecated, InnerClass.InnerAnno, Session
+
+*/
+
+package com.puppycrawl.tools.checkstyle.checks.sizes.parameternumber;
+
+public class InputParameterNumberIgnoreAnnotatedByFullyQualifiedClassName {
+    @java.lang.Deprecated
+    void method3(int a, int b, int c) {}
+
+    @Deprecated
+    void method4(int a, int b, int c) {} // violation, 'More than 2 parameters (found 3).'
```

Comment:
This needs to be discussed.
When use define `java.lang.Deprecated` he usually does not mean to use  match fully and exactly.

It can be instruction to  figure out exactly a type not a string values. We already have Checks that does this. By import we can be 100% sure on type.

Reply:
@nrmancuso , @rnveach , what are your thoughts on this?

Reply:
This might be helpful here: https://github.com/checkstyle/checkstyle/pull/14553#discussion_r1505333655

Reply:
I think we should go with most simple implementation first (what's in this PR), we can iterate on this as needed. Worst case is that some user has to put both the fully qualified name and the simple annotation name in the config.

Reply:
Do we have an update here?

Reply:
I am good to proceed as-is, @rnveach please share your thoughts

Reply:
I think we should be consistent with how we treat annotation properties like this with other checks. I don't find it friendly to have to specify both class name and FQCN.
https://checkstyle.org/checks/coding/illegaltype.html#IllegalType

Reply:
We need to make our behavior consistent with other checks and friendly to the user.

We have some check somewhere that requires FQCN and will be smart and reduce it to simple based on the import in the class.
https://checkstyle.org/checks/coding/illegaltype.html
https://checkstyle.org/checks/coding/illegalinstantiation.html

I also feel this shouldn't be delayed if it is going to create almost a breaking change in the fix.

Reply:
Do we have a resolution here?

Reply:
I agree we can extend to FQCN later on. Lets have basic implementation for now.
With current traffic of PRs I do not have time to mediate on consistency.